lib/reversetunnel/localsite.go

```diff
@@ -333,33 +335,42 @@ func (s *localSite) adviseReconnect(ctx context.Context) {
 	}
 }
 
-func (s *localSite) dialWithAgent(params DialParams) (net.Conn, error) {
-	if params.GetUserAgent == nil {
-		return nil, trace.BadParameter("user agent getter missing")
+func (s *localSite) dialAndForward(params DialParams) (_ net.Conn, retErr error) {
+	if params.GetUserAgent == nil && params.AgentlessSigner == nil {
+		return nil, trace.BadParameter("user agent getter and agentless signer both missing")
 	}
-	s.log.Debugf("Dialing with an agent from %v to %v.", params.From, params.To)
+	s.log.Debugf("Dialing and forwarding from %v to %v.", params.From, params.To)
 
-	// request user agent connection
-	userAgent, err := params.GetUserAgent()
-	if err != nil {
-		return nil, trace.Wrap(err)
+	// request user agent connection if a SSH user agent is set
+	var userAgent teleagent.Agent
+	if params.GetUserAgent != nil {
+		ua, err := params.GetUserAgent()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		userAgent = ua
```

Comment:
```suggestion
		userAgent, retErr = params.GetUserAgent()
		if retErr != nil {
			return nil, trace.Wrap(retErr)
		}
```

Reply:
That would also work, but I'll keep it as it is to avoid using `retErr` outside of a `defer`ed function.